src/bridge/crypto/rsa.cc

```diff
@@ -107,28 +134,38 @@ int Sign(int type, const unsigned char *digest_bytes,
   // conversion functions refer to some OpenSSL API functions.
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const CryptoKeyHandle *crypto_key_handle,
-      GetCryptoKeyHandleFromOpenSslRsa(rsa), false);
+      GetCryptoKeyHandleFromOpenSslRsa(rsa), 0);
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const DigestCase digest_type,
-      ConvertOpenSslNidToDigestType(type), false);
+      ConvertOpenSslNidToDigestType(type), 0);
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const std::string digest_string,
-      MakeDigestString(digest_bytes, digest_length), false);
+      MakeDigestString(digest_bytes, digest_length), 0);
 
   // Delegate handling of the signing operation to the backing layer.
   KMSENGINE_ASSIGN_OR_RETURN_WITH_OPENSSL_ERROR(
       const std::string signature,
-      crypto_key_handle->Sign(digest_type, digest_string), false);
+      crypto_key_handle->Sign(digest_type, digest_string), 0);
 
   // Copy results into the return pointers.
+  if (signature_length != nullptr) {
```

Comment:
It seems like the function should return an error if the output length pointer is null. There's no way for the caller to use `signature_return` securely if it doesn't know its length.

Reply:
And you need to make sure that `signature.length() <= INT_MAX`

Reply:
See above.